webapp/src/main/java/org/apache/atlas/web/filters/ActiveServerFilter.java

```diff
@@ -234,11 +236,28 @@ private void handleRedirect(HttpServletRequest servletRequest, HttpServletRespon
         LOG.info("Not active. Redirecting to {}", redirectLocation);
         // A POST/PUT/DELETE require special handling by sending HTTP 307 instead of the regular 301/302.
         // Reference: http://stackoverflow.com/questions/2068418/whats-the-difference-between-a-302-and-a-307-redirect
+        String sanitizedLocation = sanitizeRedirectLocation(redirectLocation);
         if (isUnsafeHttpMethod(servletRequest)) {
-            httpServletResponse.setHeader(HttpHeaders.LOCATION, redirectLocation);
+            httpServletResponse.setHeader(HttpHeaders.LOCATION, sanitizedLocation);
             httpServletResponse.setStatus(HttpServletResponse.SC_TEMPORARY_REDIRECT);
         } else {
-            httpServletResponse.sendRedirect(redirectLocation);
+            httpServletResponse.sendRedirect(sanitizedLocation);
+        }
+    }
+    public static String sanitizeRedirectLocation(String redirectLocation) {
+        if (redirectLocation == null) return null;
+        try {
+            String preProcessedUrl = redirectLocation.replace("\r", "").replace("\n", "");
+
+            preProcessedUrl = preProcessedUrl.replaceAll("%(?![0-9a-fA-F]{2})", "%25");
+
+            String encodedUrl = URLEncoder.encode(preProcessedUrl, "UTF-8");
+
+            encodedUrl = encodedUrl.replaceAll("%25([0-9a-fA-F]{2})", "%$1");
```

Comment:
This code snippet fixes incorrect % usages in URLs by replacing them with %25, the encoded form of %, and then properly encoding and correcting the entire URL. 
For example, `redirect=%unsecured%location` could misinterpret `%un` as encoded, altering URL behavior. Further snippets on line 254 and 256 involve encoding the URL to ensure compatibility and then restoring correctly encoded characters, preventing over-encoding issues like turning %25 into %2525. The final URL, `redirect=%25unsecured%25location`, ensures literal treatment of %, preventing any misinterpretation of corrupt characters like `%XY` which is not a valid hex value.

Reply:
Escape sequence removal is required to prevent `HTTP Response Splitting Attacks`